pkgs/development/compilers/symbiflow-yosys/default.nix

```
@@ -0,0 +1,111 @@
+{ stdenv
+, abc-verifier
+, bash
+, bison
+, fetchFromGitHub
+, flex
+, libffi
+, pkgconfig
+, protobuf
+, python3
+, readline
+, tcl
+, verilog
+, zlib
+, plugins ? []
+}:
+
+let
+  localAbc-verifier = abc-verifier.overrideAttrs (_: rec {
+    pname = "abc-verifier";
+    version = "2020.06.22";
+
+    src = fetchFromGitHub {
+      owner  = "YosysHQ";
+      repo   = "abc";
+      rev    = "341db25668f3054c87aa3372c794e180f629af5d";
+      sha256 = "14cgv34vz5ljkcms6nrv19vqws2hs8bgjgffk5q03cbxnm2jxv5s";
+    };
+
+    passthru.rev = src.rev;
+  });
+in
+
+stdenv.mkDerivation rec {
+  pname   = "symbiflow-yosys";
+  version = "0.8-6021-gd8b2d1a2";
+
+  src = fetchFromGitHub {
+    owner  = "SymbiFlow";
+    repo   = "yosys";
+    rev    = "d8b2d1a2b1a93057678cf49bb8f0329f191faba1";
+    sha256 = "1w8jnqzabvzixjllhb6ak2n2gmjvsn6qd996i7z70bsq5rgdkq9g";
+  };
+
+  enableParallelBuilding = true;
+  nativeBuildInputs = [ pkgconfig protobuf ];
+  buildInputs = [ tcl readline libffi python3 bison flex protobuf zlib ];
+
+  cc = if stdenv.cc.isClang then "clang" else "gcc";
+  cxx = if stdenv.cc.isClang then "clang++" else "g++";
+
+  makeFlags = [
+    "LD=${cxx}"
```

If you really want c++:

```suggestion
    "LD=${stdenv.cc.targetPrefix}c++"
```

Otherwise:

```suggestion
    "LD=${stdenv.cc.targetPrefix}ld"
```

---

done